benchmark-framework/src/main/java/io/openmessaging/benchmark/WorkloadGenerator.java

```diff
@@ -120,7 +121,15 @@ public TestResult run() throws Exception {
             }
         }
         else {
-            producerWorkAssignment.payloadData.add(payloadReader.load(workload.payloadFile));
+            File payloadFile = new File(workload.payloadFile);
```

Comment:
Really good idea, I'd only add that if it contains no files maybe just assert out with "directory was empty" or smth.